standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -327,6 +327,13 @@ private void resolveUris() throws MetaException {
       MetaStoreUtils.logAndThrowMetaException(e);
     }
 
+    if (metastoreUrisString.isEmpty() && "zookeeper".equalsIgnoreCase(serviceDiscoveryMode)) {
+      throw new MetaException("No metastore server available. "
+          + "Please ensure that at least one metastore server is online");
+    }
+
+    LOG.info("Resolved metastore uris: " + Arrays.toString(metastoreUrisString.toArray()));
```

Comment:
Nit:

Do not need to convert it to an array first.  This will work:

`LOG.info("Resolved metastore uris: {}", metastoreUrisString);`

Reply:
done